### ISSUE TITLE
convering from php

### ISSUE THREAD

**Issue by snfmanutz** [nodejs/help#3729]
### Details

I am an 85 y/o C programmer with a working page at [https://drtyoldman.w3spaces.com/] but they do not support PHP so i am using PHP and localhost with XAMMP as my platform. I want to know how i can use node.js and mysql.  I am familiar with javascript but i dont know how to interface witrh node.js.  In particular how do i execute node from my javascript html file?  Currently i use my browser, firefox, and and link to either my Javascript (html) file or my PHP file.  I use ajax (xmlhttp) in my javascript to request a mysql database table and display its content. I dont know how to do that using node.js. Do i reference node as i do jquery? Can i use the node api without starting from a command at the keyboard?  How do i get started?  here is my link to PHP, how do I substitute Node.js?

function post_it(sql) {  
  let pdata = new FormData();
  pdata.append('sql', sql);
  //console.log("at post_it: ");

  let xhr = new XMLHttpRequest();
  xhr.open('POST', **'stock.php',** true);
  xhr.onload = function () {
    let rsp = this.responseText;
    //trace("rsp="+rsp);
    //console.log("rsp="+rsp);
    process_sql_response(rsp);
  };  
  xhr.send(pdata);        
}

### Node.js version

the latest download

### Example code

function post_it(sql) {  
  let pdata = new FormData();
  pdata.append('sql', sql);
  //console.log("at post_it: ");

  let xhr = new XMLHttpRequest();
  xhr.open('POST', 'stock.php', true);
  xhr.onload = function () {
    let rsp = this.responseText;
    //trace("rsp="+rsp);
    //console.log("rsp="+rsp);
    process_sql_response(rsp);
  };  
  xhr.send(pdata);        
}

### Operating system

windows 10

### Scope

installation & linkage

### Module and version

Not applicable.

**Comment by Ayase-252** [comment]
> In particular how do i execute node from my javascript html file?

No, you can't. Node.js is server-side JavaScript runtime. There is no node.js avaliable in browser side.

> I use ajax (xmlhttp) in my javascript to request a mysql database table and display its content. I dont know how to do that using node.js. 

You could set up an API endpoint with your PHP. Database can be accessed with your PHP script. Then, request the API   from browser side with [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) or any http library you like.

**Comment by benjamingr** [comment]
> In particular how do i execute node from my javascript html file?

To clarify unlike in PHP you are not writing directly from the template engine. This is a bit like laravel and PHP frameworks where you usually have a "front-controller" pattern. You typically use a framework (like fastify) and define directly what to do when the program gets to specific routes. You can use a templating engine (like ejs) if you want to get "php like"  behavior.

There are lots of good PHP modules you may like more/less depending on whether you liked `mysqli_` or ADO as an API. Here is a module you can use https://www.npmjs.com/package/mysql .

Also be sure to read on async JavaScript and how promises and async/await work.
